includes/polling/applications/chronyd.inc.php

```diff
@@ -95,43 +101,50 @@
         $metrics['source_' . $source['source_name'] . '_' . $field] = $value;
     }
 }
-
-//
-// component processing for Chrony
-//
-$device_id = $device['device_id'];
-$options = [
-    'filter' => [
-        'device_id' => ['=', $device_id],
-        'type' => ['=', 'chronyd'],
-    ],
-];
-
-$component = new LibreNMS\Component();
-$components = $component->getComponents($device_id, $options);
-
-// if no sources, delete chrony components
-if (empty($sources)) {
-    if (isset($components[$device_id])) {
-        foreach ($components[$device_id] as $component_id => $_unused) {
-            $component->deleteComponent($component_id);
+$old_sources = $app_data['sources'];
+
+// save thge found sources
+$app_data['sources'] = $sources;
+save_app_data($app_id, $app_data);
+
+//check for added sources
+$added_sources = [];
+foreach ($sources as $source_check) {
+    $source_found = false;
+    foreach ($old_sources as $source_check2) {
+        if ($source_check == $source_check2) {
+            $source_found = true;
         }
     }
-} else {
-    if (isset($components[$device_id])) {
-        $chrony_cpnt = $components[$device_id];
-    } else {
-        $chrony_cpnt = $component->createComponent($device_id, 'chronyd');
+    if (! $source_found) {
+        $added_sources[] = $source_check;
     }
+}
 
-    // Make sure we don't readd it, just in a different order.
-    sort($sources);
-
-    $id = $component->getFirstComponentID($chrony_cpnt);
-    $chrony_cpnt[$id]['label'] = 'Chrony';
-    $chrony_cpnt[$id]['sources'] = json_encode($sources);
+//check for removed sources
+$removed_sources = [];
+foreach ($old_sources as $source_check) {
+    $source_found = false;
+    foreach ($sources as $source_check2) {
+        if ($source_check == $source_check2) {
+            $source_found = true;
+        }
+    }
+    if (! $source_found) {
+        $removed_sources[] = $source_check;
+    }
+}
 
-    $component->setComponentPrefs($device_id, $chrony_cpnt);
+// if we have any source changes, log it
+if (isset($added_sources[0]) or isset($removed_sources[0])) {
+    $log_message = 'Chronyd Source Change:';
+    if (isset($added_sources[0])) {
+        $log_message = $log_message . ' Added' . json_encode($added_sources);
+    }
+    if (isset($removed_sources[0])) {
+        $log_message = $log_message . ' Removed' . json_encode($removed_sources);
+    }
```

Comment:
When one is added and one is removed this would only catch the removed one, maybe something like this:
```
$log_message = 'Chronyd Source Change: ';
$log_message .= count($added_sources) > 0 ? ' Added ' . json_encode($added_sources) : '';
$log_message .= count($removed_sources) > 0 ? ' Removed ' . json_encode($added_sources) : '';

```

also: spaces between data and strings

Reply:
Used, thanks!